generator/src/test/java/com/graphicsfuzz/generator/transformation/donation/MakeArrayAccessesInBoundsTest.java

```diff
@@ -83,4 +83,44 @@ public void testMatrixVector2() throws Exception {
           PrettyPrinterVisitor.prettyPrintAsString(tu));
   }
 
+  @Test
+  public void testUIntConstantExprIndex() throws Exception {
+    final String shader = "#version 300 es\n"
+        + "void main() { vec3 stuff[16];"
+        + "  uint x = 19u;"
+        + "  vec3 f = stuff[x];"
+        + "}";
+    final String expected = "#version 300 es\n"
+        + "void main() { vec3 stuff[16];"
+        + "  uint x = 19u;"
+        + "  vec3 f = stuff[(x) < 16u ? x : 0u];"
+        + "}";
+    final TranslationUnit tu = ParseHelper.parse(shader);
+    final Typer typer = new Typer(tu);
+    MakeArrayAccessesInBounds.makeInBounds(tu, typer);
+    assertEquals(PrettyPrinterVisitor.prettyPrintAsString(ParseHelper.parse(expected)),
+        PrettyPrinterVisitor.prettyPrintAsString(tu));
+  }
+
+  @Test
+  public void testUIntFunctionCallReturnIndex() throws Exception {
```

Comment:
Can you change the shading language version, both before and after, to "#version 310 es"?  This is because uaddCarry does not exist before "300 es".

Reply:
Addressed.